quic/s2n-quic-core/src/datagram/default.rs

```diff
@@ -241,13 +238,13 @@ pub struct Datagram {
 
 #[non_exhaustive]
 #[derive(Debug, PartialEq)]
-pub enum SendDatagramError {
+pub enum DatagramError {
     QueueAtCapacity,
     ExceedsPeerTransportLimits,
     ConnectionError { error: connection::Error },
```

Comment:
All of the variants should ideally be non-exhaustive as well:

```suggestion
    #[non_exhaustive]
    QueueAtCapacity,
    #[non_exhaustive]
    ExceedsPeerTransportLimits,
    #[non_exhaustive]
    ConnectionError { error: connection::Error },
```